### PR TITLE
Fix a few (but not all) too-complex expressions that won't fail properly with just "set -e"

### PR DESCRIPTION
**Pull request by tianon**

### CODE REVIEW

File: bashbrew/bashbrew.sh [hunk 1]
```diff
@@ -280,6 +278,22 @@ for repoTag in "${repos[@]}"; do
...
+		cd "$1"
+		git reset -q HEAD
+		git checkout -q -- .
+		git clean -dfxq
+		git checkout -q "$2" --
+		cd "$1/$3"
+		"$dir/git-set-mtimes"
+	)
+	return 0
+}
```
Comment (tianon) [inline bashbrew/bashbrew.sh]: This one still won't "fail" properly, but at least it's abstracted for better readability.

**Comment by md5** [comment]
Would `-o pipefail` help with any of these?

**Comment by tianon** [comment]
Nope, unfortunately not. :cry:

https://github.com/docker/docker/pull/12500 actually has quotes and links to the relevant Bash documentation for where `set -e` falls on its face, which makes me a really sad panda (the gist being that it falls on its face any time an expression gets "complicated" -- inside expressions including `&&`, `||`, `if` context, etc).

**Comment by md5** [comment]
Gotta work with what you're given I suppose.

**Comment by tianon** [comment]
Yeah, I still love Bash, but this knowledge put a serious dent in my Bash-as-a-serious-programming-language-contender argument. :cry: :disappointed:

**Comment by yosifkit** [comment]
LGTM

**Pull request merged**